services/src/main/java/org/keycloak/email/freemarker/FreeMarkerEmailTemplateProvider.java

```diff
@@ -149,7 +149,7 @@ public void sendConfirmIdentityBrokerLink(String link, long expirationInMinutes)
         String idpDisplayName = brokerContext.getIdpConfig().getDisplayName();
         idpAlias = ObjectUtil.capitalize(idpAlias);
 
-        if (idpDisplayName != null) {
+        if (idpDisplayName != null && idpDisplayName.length() > 0) {
```

Comment:
Just nitpick. What about using this particular method?

```suggestion
        if (StringUtil.isNotBlank(idpDisplayName) {
```